tests/App/config/services.yml

```diff
@@ -42,10 +42,14 @@ services:
         class: Sonata\AdminBundle\Tests\App\Datagrid\ProxyQuery
 
     Sonata\AdminBundle\Tests\App\Admin\FooAdmin:
-        arguments: [~, Sonata\AdminBundle\Tests\App\Model\Foo, Sonata\AdminBundle\Controller\CRUDController]
+        arguments: [~, Sonata\AdminBundle\Tests\App\Model\Foo, 'sonata.admin.controller.crud']
         tags:
             - {name: sonata.admin, manager_type: test, label: Foo}
 
+    Sonata\AdminBundle\Tests\App\Controller\CustomCRUDController:
+        public: true
```

Comment:
The recommended symfony config is 
```
services:
    _defaults:
        autowire: true
        autoconfigure: true
        public: false

    # makes classes in src/ available to be used as services
    # this creates a service per class whose id is the fully-qualified class name
    App\:
        resource: ../src/*
        exclude: ../src/{Entity,Migrations,Kernel.php}

    # controllers are imported separately to make sure services can be injected
    # as action arguments even if you don't extend any base controller class
    App\Controller\:
        resource: ../src/Controller
        tags: [controller.service_arguments]
```
(See https://github.com/symfony/demo/blob/main/config/services.yaml)

So CRUDController will be autowire by default, but will be private too.
Do we have another way than asking them to put all the CRUDController public ?

Reply:
Yes, I'll change it, I was just trying. BTW controllers are not private with this configuration, adding the `controller.service_arguments` makes them public.